lib/src/arithmetic/floating_point/floating_point_value.dart

```diff
@@ -907,4 +904,117 @@ class FloatingPoint8Value extends FloatingPointValue {
+      egressExponent = computeMaxExponent(egressExponentWidth) +
+          computeBias(egressExponentWidth) +
+          1;
+      egressMantissa <= LogicValue.zero(egressMantissaWidth);
+
+      return FloatingPointValue(
+          sign: egressSign, exponent: egressExponent, mantissa: egressMantissa);
+    }
+
+    // Zero cases
+    if (ingress_fpVal.isZero()) {
+      return FloatingPointValue(
+          sign: egressSign,
+          exponent: LogicValue.zero(egressExponentWidth),
+          mantissa: LogicValue.zero(egressMantissaWidth));
+    }
+
+    // Subnormal cases
+    // In subnormal cases, the ingress exponent is all zeroes
+    // So we are really only concerned about the mantissa
+    if (ingress_fpVal.isSubnormal()) {
+      if (egressMantissa < ingress_fpVal.mantissa) {
+        var mantissaWidthDiff =
+            egressMantissa.width - ingress_fpVal.mantissa.width;
+        var mask = ingress_fpVal.mantissa &
+            LogicValue.filled(mantissaWidthDiff, LogicValue.one);
+
+        // The lower bits of the mantissa and extracted and rounded using the sticky bit from the mask
+        var lowMantissaBits =
+            (ingress_fpVal.mantissa >>> mantissaWidthDiff) | (mask != 0);
+        exgressExponent =
+            LogicValue.filled(egressExponentWidth, LogicValue.zero);
+      } else if (egressMantissa > ingress_fpVal.mantissa) {
+        // Normalize the mantissa by counting the leading zeroes until it hit a non-zero bit
+        int leadingZeroes = countLeadingZeroes(ingress_fpVal.mantissa);
+        egressMantissa = ingress_fpVal.mantissa >>> leadingZeroes;
+        egressExponent = ingress_fpVal.exponent - leadingZeroes;
+      } else if (egressMantissa == ingress_fpVal.mantissa) {}
+    }
+
+    // Normal cases
+
+    if (ingress_fpVal.mantissa.width <= egressMantissaWidth) {
+      // Zero pad the mantissa in the case of precision gain
+      egressMantissa = ingress_fpVal.mantissa <<
+          (egressMantissaWidth - ingress_fpVal.mantissa.width);
+    } else {
+      // Trim the mantissa in the case of precision loss from the right end, preserving the left end
+      egressMantissa = ingress_fpVal.mantissa.slice(
+          ingress_fpVal.mantissa.width - 1,
+          egressMantissaWidth - ingress_fpVal.mantissa.width);
+    }
+
+    egressExponent = (ingress_fpVal.exponent.toInt() - ingress_fpVal.bias()) +
+        FloatingPointValue.computeBias(egressExponentWidth);
+  }
+}
+
+/// Count the number of leading zeroes in a LogicValue.
+///
+/// This is a binary count of the number of leading zeroes in the
+/// [LogicValue].
+///
+/// The logic value is treated as a binary number, with the rightmost
+/// bit being the least significant bit (bit 0). The leading zeroes
+/// are counted from the most significant bit (bit width-1) towards
+/// the least significant bit.
+///
+/// For example, the binary number 0b00101010 would return 3, as
+/// there are 3 leading zeroes.
+int countLeadingZeroes(LogicValue value) {
```

Comment:
if this is a good utility in a generic sense, maybe it should be an extension on `LogicValue` or even added to ROHD itself.  If it's mostly just useful for this math stuff, let's keep it private (start name with `_`) so that it's not exposed as a public API